### ISSUE TITLE
CQLSH only appears to require certfile for TLS connections

### ISSUE THREAD

**Issue by aegershman** [yugabyte/yugabyte-db#3846]
After enabling client-to-server encryption, I would like to use `cqlsh` with `ssl`.

[The documentation says these env vars must be set:](https://docs.yugabyte.com/latest/secure/tls-encryption/connect-to-cluster/)

```sh
$ export SSL_CERTFILE=<path to file>/ca.crt
$ export SSL_USERCERT=<path to file>/node.<name>.crt
$ export SSL_USERKEY=<path to file>/node.<name>.key
```

There are two small things I'd like to report / possibly clarify?

- I'm finding that TLS connection still works with _only_ `SSL_CERTFILE` set, and not needing `SSL_USERCERT` or `SSL_USERKEY` to be set.

- [I'm finding that using `.cqlshrc` works as well](https://github.com/apache/cassandra/blob/trunk/conf/cqlshrc.sample#L103), e.g. passing `yugabyte/bin/cqlsh --cqlshrc /somedir/cqlshrc`

```conf
[connection]
hostname = unimportant-etc
ssl = true

[ssl]
certfile = /somedir/certs/ca.crt
```

but interestingly enough, in `.cqlshrc`, specifying `usercert` and `userkey` (the presumed analogies of `SSL_USERCERT` and `SSL_USERKEY`) like so:

```conf
[ssl]
certfile = /somedir/certs/ca.crt
usercert = /somedir/certs/node.common.crt
userkey = /somedir/certs/node.common.key
```

results in an error looking like this:

```
stdout | Connection error: ('Unable to connect to any servers', {'10.156.89.143': error(8, "Tried connecting to [('10.156.89.143', 9042)]. Last error: _ssl.c:507: EOF occurred in violation of protocol")})
```

But when only specifying `certfile = /somedir/certs/ca.crt`, it connects and works just fine.

It seems like _either_ `SSL_USERCERT` and `SSL_USERKEY` aren't necessary (?), or that there's a discrepancy in how TLS certs are used between env vars and `.cqlshrc`

So, long story short, I guess for now we can get away with only specifying the certfile in `cqlshrc`.

Thanks for any consideration! Maybe I'm way off, I'll continue down this path and if I see that I've completely gotten this wrong I'll absolutely post an update. **It's entirely possible that I'm completely using this wrong**, and I apologize I don't want to waste anyone's time or energy

**Comment by iSignal** [comment]
This is expected behavior from the code and our docs at https://docs.yugabyte.com/latest/secure/tls-encryption/connect-to-cluster/ have been fixed to reflect this. Sorry for the confusion. Thanks for reporting @aegershman !
